### PR TITLE
chore: upgrade rrweb to alpha.17

### PR DESCRIPTION
**Pull request by daibhin**
## Changes

Following the big bump in https://github.com/PostHog/posthog-js/pull/1276 which seems to have rolled out fine it is probably now worth moving to the latest version which has no scary changes.

Reapplied all patches

Best new feature is `dialog` element support

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
         "@rollup/plugin-node-resolve": "^15.3.0",
         "@rollup/plugin-terser": "^0.4.4",
         "@rollup/plugin-typescript": "^12.1.1",
-        "@rrweb/types": "2.0.0-alpha.16",
+        "@rrweb/rrweb-plugin-console-record": "2.0.0-alpha.15",
```
[the remark this answers is not in the document]
Comment (daibhin) [reply package.json]: Yep, good catch. Really don't know what I missed there. Update in the next commit

[the remark this answers is not in the document]
Comment (pauldambra) [reply package.json]: i'll never forgive you :p

File: patches/rrweb@2.0.0-alpha.17.patch [hunk 2]
```diff
@@ -22,15 +22,15 @@ index 1db2d08b2fe9a73521001c3aa88b1a64d1b3e71a..5a3ea8410b0f1dfa7248b9cb9752c04e
 +    const href = n2.href;
 +    let stylesheet = findStylesheet(doc, href);
 +    if (!stylesheet && href.includes('.css')) {
-+      const rootDomain = window.location.origin
-+      const stylesheetPath = href.replace(window.location.href, '')
-+      const potentialStylesheetHref = rootDomain + '/' + stylesheetPath
++      const rootDomain = window.location.origin;
++      const stylesheetPath = href.replace(window.location.href, '');
++      const potentialStylesheetHref = rootDomain + '/' + stylesheetPath;
```
Comment (daibhin) [inline patches/rrweb@2.0.0-alpha.17.patch:27]: No real difference here, just added `;` at the end of each line

**Comment by pauldambra** [comment]
Screenshot 2024-10-22 at 18 40 14
holy cow

**Comment by daibhin** [comment]
> Screenshot 2024-10-22 at 18 40 14 holy cow

Nooooiceeee 😎 

----

I'm not sure what's causing this. Here is the list of changes between 16 and 17:

Screenshot 2024-10-22 at 18 43 45


Perhaps all of postcss is being pulled in? It should only be needed during playback

**Comment by pauldambra** [comment]
> Perhaps all of postcss is being pulled in? It should only be needed during playback

that's my first thought too

**Review by marandaneto — APPROVED** [review]

**Review by pauldambra — APPROVED** [review]
weeeellllll, do i approve this or not.

it's way tidier since we lose the patches
it works locally, which is the definitive test to prove all is well
the bundled file is waaaaayyyyy bigger which would be nice to avoid

**Comment by pauldambra** [comment]
yep @daibhin comparing v16 and v17 built output includes things like

```
.NODE_ENV&&"CssSyntaxError"===e.name&&t&&t.from&&(/\.scss$/i.test(t.from)?e.message+="\nYou tried to parse SCSS with the standard CSS parser; try again with the postcss-scss parser":/\.sass/i.test(t.from)?e.message+="\nYou tried to parse Sass with the standard CSS parser; try again with the postcss-sass parser":/\
```

so pretty sure that postcss is being included

**Comment by pauldambra** [comment]
but then the console recording plugin is as big as rrweb when looking at the bundle
Screenshot 2024-10-22 at 21 08 43

so, 🤷

**Comment by daibhin** [comment]
Closing in favour of https://github.com/PostHog/posthog-js/pull/1489

**Pull request closed without merging**